syntax = "proto3";

package protos;

service TurnBasedServer {
  rpc GetInitialObservation(SubEnvInfo) returns (InitialObservation) {}
  rpc GetObservation(ActionInfo) returns (Observation) {}
}

message ActionInfo {
  SubEnvInfo sub_env_info = 1;
  NDArray action = 2;
}

message Observation {
  NDArray observation = 1;
  float reward = 2;
  bool done = 3;
  string info = 4;
}

message InitialObservation {
  NDArray observation = 1;
}

message NDArray {
  // From here: https://github.com/xainag/numproto/blob/master/numproto/protobuf/ndarray.proto
  bytes ndarray = 1;
}

message SubEnvInfo {
  string sub_env_id = 1;
}